internal/resource/config/authenticationselector/authentication_selector_resource.go

```diff
@@ -0,0 +1,349 @@
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func (r *authenticationSelectorResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	var plan, state *authenticationSelectorResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	var respDiags diag.Diagnostics
+
+	if plan == nil || state == nil {
+		return
+	}
+
+	plan.Configuration, respDiags = pluginconfiguration.MarkComputedAttrsUnknownOnChange(plan.Configuration, state.Configuration)
+	resp.Diagnostics.Append(respDiags...)
+
+	resp.Plan.Set(ctx, plan)
+}
+
+func readAuthenticationSelectorsResponse(ctx context.Context, r *client.AuthenticationSelector, state *authenticationSelectorResourceModel, configurationFromPlan types.Object) diag.Diagnostics {
+	var diags, objDiags diag.Diagnostics
+
+	state.LastModified = types.StringValue(r.LastModified.Format(time.RFC3339))
+	state.AttributeContract, objDiags = types.ObjectValueFrom(ctx, attributeContractAttrType, r.AttributeContract)
+	diags = append(diags, objDiags...)
+	state.SelectorId = types.StringValue(r.Id)
+	state.Id = types.StringValue(r.Id)
+	state.Name = types.StringValue(r.Name)
+	state.PluginDescriptorRef, objDiags = resourcelink.ToStateNoLocation(&r.PluginDescriptorRef)
+	diags = append(diags, objDiags...)
+	state.ParentRef, objDiags = resourcelink.ToStateNoLocation(r.ParentRef)
+	diags = append(diags, objDiags...)
+	state.Configuration, objDiags = pluginconfiguration.ToState(configurationFromPlan, &r.Configuration)
+	diags = append(diags, objDiags...)
+
+	// make sure all object type building appends diags
+	return diags
+}
+
+func (r *authenticationSelectorResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan authenticationSelectorResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var hasObjectErrMap = make(map[error]bool)
+	pluginDescriptorRef, err := resourcelink.ClientStruct(plan.PluginDescriptorRef)
+	if err != nil {
+		hasObjectErrMap[err] = true
+	}
+
+	var configuration client.PluginConfiguration
+	err = json.Unmarshal([]byte(internaljson.FromValue(plan.Configuration, false)), &configuration)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to read configuration from plan", err.Error())
+		return
+	}
+
+	for err, hasErr := range hasObjectErrMap {
+		if hasErr {
+			resp.Diagnostics.AddError("Failed to create an Authentication Selector due to dependent object", err.Error())
+		}
+	}
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createAuthenticationSelectors := client.NewAuthenticationSelector(plan.SelectorId.ValueString(), plan.Name.ValueString(), *pluginDescriptorRef, configuration)
```

Comment:
We haven't done this for other resources yet but we could start using a common createAuthenticationSelectorRequest (or something) method that could reduce a little bit of duplication between Create and Update for building the required parameters.

Reply:
I've added a JIRA case for this